### PR TITLE
WIP: Allowing SSL connections to Mongo

### PR DESCRIPTION
**Pull request by arschles**
This PR also refactors how mongo connection details are passed around.

Closes #216

### CODE REVIEW

**Comment by arschles** [comment]
This is failing because `github.com/globalsign/mgo` doesn't seem to connect properly to mongo when I put a port in the slice of addresses in `mgo.DialInfo`. still needs investigation

File: pkg/config/env/mongo.go [hunk 1]
```diff
@@ -67,3 +72,52 @@ func MongoConnectionTimeoutWithDefault(value string) string {
...
+	timeoutStr := MongoConnectionTimeoutWithDefault("1")
+	timeoutSec, err := strconv.Atoi(timeoutStr)
+	if err != nil {
+		multierror.Append(errs, errors.WithMessage(err, "invalid value for timeout"))
+	}
+	timeout := time.Duration(timeoutSec) * time.Second
+	sslStr := MongoSSLWithDefault("true")
+	ssl, err := strconv.ParseBool(sslStr)
+	if err != nil {
+		multierror.Append(errs, errors.WithMessage(err, "invalid value for mongo SSL"))
```
Comment (michalpristas) [inline pkg/config/env/mongo.go]: I ❤️ the multierror approach, ain't nothing worse than resolve-deploy cycle 5 times learning missing value one at a time

File: pkg/storage/mongo/conn/conn.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	dialInfo := &mgo.DialInfo{
+		Addrs:    []string{fmt.Sprintf("%s:%d", deets.Host, deets.Port)},
+		Timeout:  deets.Timeout,
+		Database: db,
+		Username: deets.User,
+		Password: deets.Password,
+	}
+	if deets.SSL {
+		dialInfo.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
+			return tls.Dial("tcp", addr.String(), &tls.Config{})
```
Comment (michalpristas) [inline pkg/storage/mongo/conn/conn.go]: todo for certs?

**Comment by michalpristas** [comment]
@arschles working on this? do you need any help?

**Comment by michalpristas** [comment]
i may try to take over, let's see what comes up 👍

**Comment by michalpristas** [comment]
I have this PR #527 in, it's not so pretty as this one, I started from scratch and tested it locally. I will address comments there as in this PR there is some rebasing piling up

**Pull request closed without merging**